Assets/HoloToolkit/Utilities/Scripts/Performance/PerformanceCounters.cs

```diff
@@ -0,0 +1,265 @@
+                await FileIO.WriteTextAsync(storage, "Time, FrameTime, FrameGpuTime\r\n");
+            }
+
+            await FileIO.AppendTextAsync(storage, String.Join("", timeSamples));
+        }
+#endif
+#if STORE_APP_BINARY
+        private string storageFileName = null;
+        private Task backgroundFlush = null;
+        private StorageFile storage = null;
+#endif
+        }
+
+        public bool GpuTimeEnabled
+        {
+            get
+            {
+                return profileGpuTime;
+            }
+
+            set
+            {
+                profileGpuTime = value;
+                if (!profileGpuTime)
+                {
+                    LastFrameGpuTime = 0;
+                    gpuTimeAccumulator = 0;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The amount of time it took to complete the last frame
+        /// </summary>
+        public float LastFrameTime { get; private set; }
+        /// <summary>
+        /// The amount of time it took to render the last frame
+        /// </summary>
+        public float LastFrameGpuTime { get; private set; }
+
+        /// <summary>
+        /// The native frame rate as determined by the display refresh rate
+        /// </summary>
+        public int TargetFrameRate { get; private set; }
+
+        [SerializeField]
+        [Tooltip("The number of frames over which the performance counters are averaged")]
+        private int averagingInvterval = 10;
+
+        /// <summary>
+        /// The number of samples that need to collect before saving performance
+        /// data to the file
+        /// </summary>
+        private int saveDataInterval = 100;
+
+        [SerializeField]
+        [Tooltip("Save the performance date to a CSV file")]
+        private bool dumpPerformanceData = false;
+
+        [SerializeField]
+        [Tooltip("Enable GPU timing performance counter")]
+        private bool profileGpuTime = false;
+
+        [SerializeField]
+        [Tooltip("Main HMD camera. Needed for GpuTiming")]
+        private Camera hmdCamera;
+
+        private GpuTimingCamera gpuCamera;
+
+        private float lastSampleStartTime = 0;
+        private float lastFrameStartTime = 0;
+        private float gpuTimeAccumulator = 0;
+        private int currentFrame = 0;
+        private int currentSample = 0;
+
+        private TimeSampleStorage sampleStorage = null;
+        private LinkedList<TimeSample> timeSamples = new LinkedList<TimeSample>();
+
+        protected override void InitializeInternal()
+        {
+            TargetFrameRate = (int)UnityEngine.XR.XRDevice.refreshRate;
+
+            if (TargetFrameRate == 0)
+            {
+                TargetFrameRate = 60;
+                if (!Application.isEditor)
+                {
+                    Debug.LogWarning("Could not retrieve the HMD's native refresh rate. Assuming " + TargetFrameRate + " Hz.");
```

Comment:
Should we always log this if we're not in the Editor?

Reply:
why not!

Reply:
Just seems like it would always be the case, not the exception.

Reply:
Also, logs do incur some overhead, and this one is a warning, making me think there's something wrong.